client/components/deposits-overview/recent-deposits-list.tsx

```diff
@@ -24,26 +24,26 @@ import { CachedDeposit } from 'wcpay/types/deposits';
 import { formatCurrency } from 'wcpay/utils/currency';
 import { getDetailsURL } from 'wcpay/components/details-link';
 
-interface DepositRowProps {
-	deposit: CachedDeposit;
-}
-
 interface RecentDepositsProps {
 	deposits: CachedDeposit[];
 }
 
 const tableClass = 'wcpay-deposits-overview__table';
 
 /**
- * Renders a recent deposits table row.
+ * Renders the Recent Deposit list component.
  *
- * @return {JSX.Element} Deposit table row.
+ * This component includes the recent deposit heading, table and notice.
  */
-const DepositTableRow: React.FC< DepositRowProps > = ( {
-	deposit,
-} ): JSX.Element => {
-	return (
-		<Flex className={ `${ tableClass }__row` }>
+const RecentDepositsList: React.FC< RecentDepositsProps > = ( {
```

Comment:
Curious about your thoughts on inlining `RecentDepositsProps`, so the component declaration includes info on the type of props to pass in (i.e. more self-contained). 

At face value I prefer inline but there might be good reasons for separate type. I want to get aligned with best practice.

Reply:
Since this is one prop, inlining would be tidy and I'd be OK with either.

I lean towards defining a separate type, as it becomes easier to read as more props or descriptions are added.

```tsx
// Single prop, easy to read ✅
const RecentDepositsList: React.FC< {
	deposits: CachedDeposit[];
} > = ( {
 	deposits,
} ) => {
```

```tsx
// Single prop with description, still good to read ✅
const RecentDepositsList: React.FC< {
	/** Array of deposits to render in the RecentDepositsList **/
	deposits: CachedDeposit[];
} > = ( {
 	deposits,
} ) => {
```


```tsx
// Multiple props with descriptions, harder to read 🟡 **/
const RecentDepositsList: React.FC< {
	/** Array of deposits to render in the RecentDepositsList **/
	deposits: CachedDeposit[];
	/** Another prop **/
	isLoading?: boolean;
	/** Function **/
	onChange?: ( name: string, enabled: boolean ) => void;
	/** Another function **/
	onKeyDown?: ( event: React.KeyboardEvent< HTMLInputElement > ) => void;
} > = ( {
 	deposits,
	isLoading,
	onChange,
	onKeyDown,
} ) => {
```

```tsx
// Separate multiple props with descriptions, easier to read ✅ **/
interface RecentDepositProps {
	/** Array of deposits to render in the RecentDepositsList **/
	deposits: CachedDeposit[];
	/** Another prop **/
	isLoading?: boolean;
	/** Function **/
	onChange?: ( name: string, enabled: boolean ) => void;
	/** Another function **/
	onKeyDown?: ( event: React.KeyboardEvent< HTMLInputElement > ) => void;
};

const RecentDepositsList: React.FC< RecentDepositProps > = ( {
 	deposits,
	isLoading,
	onChange,
	onKeyDown,
} ) => {
```

> i.e. more self-contained).

I see the entire React component file as a self-contained component, including imports and any functions defined within the file.

If the Props interface was defined outside the file, I'd prefer to move it inline to the file, to make it self-contained.